app/pages/admin/project-status.jsx

```diff
@@ -90,8 +132,33 @@ class ProjectStatus extends Component {
   getWorkflows() {
     const fields = 'display_name,active,configuration,grouped,prioritized,retirement';
     return getWorkflowsInOrder(this.state.project, { fields }).then((workflows) => {
+
+      // Setup transitional input values
+      const valTrainingSetIds = {}, valTrainingChances = {}, valTrainingDefaultChance = {}
+      workflows.forEach((wf) => {
+        valTrainingSetIds[wf.id] =
+          (wf.configuration.training_set_ids && wf.configuration.training_set_ids.join)
+          ? wf.configuration.training_set_ids.join(',')
+          : '';
+        valTrainingChances[wf.id] =
+          (wf.configuration.training_chances && wf.configuration.training_chances.join)
+          ? wf.configuration.training_chances.join(',')
+          : '';
+        // Note: please allow default chance to be 0. This is a valid edge case.
+        valTrainingDefaultChance[wf.id] =
+          (wf.configuration.training_default_chance >= 0 && wf.configuration.training_default_chance !== null)
```

Comment:
I think that `>= 0` means that the default chance is never null for the second test? So I'm not sure what the second test is testing?

Reply:
omfg you won't believe this one...

```
undefined >= 0 (OR) undefined <= 0 (OR) undefined == 0
result: false

"" >= 0 (OR) "" <= 0 (OR) "" == 0
result: true

null >= 0 (OR) null <= 0
result true

null == 0 (OR) null > 0 (OR) null < 0
result false
```

🤦

Reply:
Oh no!

![Oh no, facepalm GIF](https://media.giphy.com/media/LpQuxhwDhzLCEKVYFh/giphy-downsized.gif)

Reply:
In happier news, the [`??` operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator) might work here.

Reply:
`wf.configuration.training_default_chance ?? ''` will be `0` if the default chance is zero, but an empty string if the default chance is `null`.